src/main/java/it/pagopa/ecommerce/commons/client/NpgClient.java

```diff
@@ -321,10 +323,71 @@ public Mono<FieldsDto> buildForm(
                 paymentMethod,
                 defaultApiKey,
                 contractId,
+                null,
                 null
         );
     }
 
+    /**
+     * method to invoke the orders/build api in order to start a payment session for
+     * subsequent payment, retrieve the sessionId and state. This method ensures
+     * that the request dto for the orders/build api will be built in the right way
+     * (it is easy to build it manually with wrong values, e.g. <i>amount</i> or
+     * <i>currency</i> as a string can be easily confused).
+     *
+     * @param correlationId   the unique id to identify the rest api invocation
+     * @param merchantUrl     the merchant url of the payment session
+     * @param resultUrl       the result url where the user should be redirected at
+     *                        the end of the payment session
+     * @param notificationUrl the notification url where notify the session
+     * @param cancelUrl       the url where the user should be redirected if the
+     *                        session is canceled by the user
+     * @param orderId         the orderId of the payment session
+     * @param customerId      the customerId url of the api
+     * @param paymentMethod   the payment method for which the form should be built
+     * @param defaultApiKey   default API key
+     * @param contractId      the wallet contractId
+     * @param language        the language chosen by the user
+     * @return An object containing sessionId and state
+     */
+    /*
+     * @formatter:off
+     *
+     * Warning java:S107 - Methods should not have too many parameters
+     * Suppressed because this method wraps the underlying API which has this many parameters
+     *
+     * @formatter:on
+     */
+    @SuppressWarnings("java:S107")
+    public Mono<FieldsDto> buildForm(
+                                     @NotNull UUID correlationId,
+                                     @NotNull URI merchantUrl,
+                                     @NotNull URI resultUrl,
+                                     @NotNull URI notificationUrl,
+                                     @NotNull URI cancelUrl,
+                                     @NotNull String orderId,
+                                     @NotNull String customerId,
+                                     @NonNull PaymentMethod paymentMethod,
+                                     @NonNull String defaultApiKey,
+                                     String contractId,
+                                     String language
```

Comment:
Add the `@Nullable` annotation if they can be `null` :)

```suggestion
                                     @Nullable String contractId,
                                     @Nullable String language
```

Reply:
done